Parse/src/main/java/com/parse/ParseAWSRequest.java

```diff
@@ -19,14 +19,18 @@
  * Request returns a byte array of the response and provides a callback the progress of the data
  * read from the network.
  */
-/** package */ class ParseAWSRequest extends ParseRequest<byte[]> {
+/** package */ class ParseAWSRequest extends ParseRequest<Void> {
 
-  public ParseAWSRequest(Method method, String url) {
+  // The temp file is used to save the ParseFile content when we fetch it from server
+  private final File tempFile;
+
+  public ParseAWSRequest(Method method, String url, File tempFile) {
     super(method, url);
+    this.tempFile = tempFile;
   }
 
   @Override
-  protected Task<byte[]> onResponseAsync(ParseHttpResponse response,
+  protected Task<Void> onResponseAsync(ParseHttpResponse response,
```

Comment:
We should probably do the operations in this method on `ParseExecutors.io()` since we're doing disk i/o now or else we'd be on `ParseRequest.NETWORK_EXECUTOR` by default.